InTheHand.Net.Obex/ObexWebRequest.cs

```diff
@@ -1038,8 +1050,16 @@ public override WebResponse GetResponse()
             ObexStatusCode status;
             MemoryStream ms = new MemoryStream();
             WebHeaderCollection responseHeaders = new WebHeaderCollection();
+            // Added David Rodgers to facilitate sending an Md5 checksum in the header
+            // could be expanded to loop through Headers and add any headers that are set by the user
+            if (Headers["ContentMd5"] != null)
```

Comment:
Do you need to add the existing request header here? - because the responseHeaders should contain the headers as sent by the remote device along with the retrieved file.

Reply:
When sending the file I set the user defined header request.Headers.Add("ContentMd5", md5); ("ContentMd5" would change to "User0" with your new changes) before calling request.GetResponse(); The call to GetResponse() then creates responseHeaders as a new WebHeaderCollection() instance effectively wiping out the previously set user defined header. Other headers are added in ObexWebRequest:DoPut(); With your new user defined headers where would we set them when sending a file?

Reply:
I will wait until I find out how to set a user defined header so I can implement and test before resubmitting.

Reply:
You have not commented about public ObexListenerContext(Stream rs) with a stream parameter instead of socket, what are your thoughts about it?

Reply:
Sorry I've been distracted and forgot to follow up on this. Now that the listener can work from either a stream or a socket, I don't think the public constructor is needed because this can be created through the normal listener flow.

Reply:
I can see where you test for socket == null in ObexListenerContext at line 257 and assign stream on line 259 but this is not used. Line 283 says "// call to stream implementation to go here" so there is no ObexListenerContext returned. So this does not seem to be any further ahead with regard to getting the ObexListenerContext "through the normal listener flow".  My other issue is that I have to use BluetoothListener rather than ObexListener because I need to set a custom service GUID. It would be great if we could pass a custom GUID to the BluetoothListener constructor.  This is why I created the public ObexListenerContext constructor which took a stream argument. I want to use ObexListenerContext but also need a custom service GUID.

Reply:
Good point, then it should expose the public constructor for those cases. There is definitely scope for more flexibility in the listener so it can support custom services

Reply:
Will you be making changes to support custom services and when would you expect to include them?

Reply:
I think based on time available to this at the moment we assume that it won't happen soon. It'll possibly require work to both the BluetoothListener and ObexListener to fully support.